### DISCUSSION TITLE
Context based Indent Guide

### DISCUSSION THREAD

**Discussion by duduribeiro** [zed-industries/zed#36945] [Feature Requests]
### Check for existing issues

- [x] Completed

### Describe the feature

(tried to find an existent issue and did not find anything related)

Hey folks, is there any way to configure indent guides to be shown only on the current LSP or Treesitter context? 

For example if I want to show the indent guides only for the current method I am focusing. 

I tried to set 

    "line_width": 0,
but it accepts a greater or equal 1 😄 

What I want to achieve is something like this on my nvim config:

See that when I focus on the method it shows the indent guide for it
![Image](https://github.com/user-attachments/assets/cbe1c4b3-3da1-4cc5-8b88-fd2db586d941)

and when I focus on another thing it changes the indent guide

![Image](https://github.com/user-attachments/assets/35d1f107-d350-4050-b471-f92317bf49ee)

is this something feasible to do in Zed?

thanks a lot

**Comment by notpeter** [comment]
This is not currently supported. I imagine you could get something like this via tree-sitter.  You might look at the implementation of `editor::SelectLargerSyntaxNode` for an example of how to this. In thinking about how to track the state, I assume on every cursor movement you would need to calculate the current indent context and then create an alternate implementation for drawing `indent_guides` which iterated over all visible cursors and drew it appropriately.

It's a nifty idea, but not something we're looking to immediately prioritize.
Thanks for filing.
